flag_engine/environments/schemas.py

```diff
@@ -11,9 +11,8 @@
 class EnvironmentSchema(LoadToModelSchema):
     id = fields.Int()
     api_key = fields.Str()
-    feature_states = ListOrDjangoRelatedManagerField(
-        fields.Nested(FeatureStateSchema),
-        metadata={"filter_kwargs": {"feature_segment_id": None, "identity_id": None}},
+    _all_feature_states = ListOrDjangoRelatedManagerField(
```

Comment:
there is a small problem with this approach: since `_all_feature_states` does not exists on django model we have to tell marshmallow to use `feature_states` to populate this field(that can be done with `attribute`) once we do that dump works fine, but since load does not(because of the `attribute` it returns a dictionary with `feature_states` as one of the key instead of `_all_feature_states` ) to solve that either we can override post_load use two different schemas one for loading and one for dumping.
ref: https://github.com/marshmallow-code/marshmallow/issues/791

Reply:
Ah yeah, that makes sense. I do feel like maybe we're battling against marshmallow a little and maybe using 2 separate schemas (superclassing a base schema for any common attributes) would make things a lot cleaner. What do you think?